docs/developers-guide.md

```diff
@@ -3,319 +3,131 @@
+        - [Kubernetes cluster](#kubernetes-cluster)
+    - [Contribution](#contribution)
+    - [Development](#development)
+        - [Build openstack-cloud-controller-manager image](#build-openstack-cloud-controller-manager-image)
+        - [Troubleshooting](#troubleshooting)
+        - [Review process](#review-process)
 
 <!-- END doctoc generated TOC please keep comment here to allow auto update -->
 
 # Getting Started with Cloud Provider OpenStack Development
 
-This guide will help you get started with building a development environment for you
-to build and run a single node Kubernetes cluster with the OpenStack Cloud Provider
+This guide will help you get started with building a development environment run a single node Kubernetes cluster with the OpenStack Cloud Provider
 enabled.
 
-## Prerequisites
 
-To get started, you will need to set up your development environment.
+## Prerequisites
 
 ### OpenStack Cloud
 You will need access to an OpenStack cloud, either public or private. You can sign up
 for a public OpenStack Cloud through the [OpenStack Passport](https://www.openstack.org/passport)
 program, or you can install a small private development cloud with
 [DevStack](https://docs.openstack.org/devstack/latest/).
 
-Once you have obtained access to an OpenStack cloud, you will need to start a development
-VM. The rest of this guide assumes a CentOS 7 cloud image, but should be easily transferrable
-to whatever development environment you prefer. You will need to have your cloud credentials
-loaded into your environment. For example, I use this `openrc` file:
-
-```
-export OS_PROJECT_DOMAIN_NAME=Default
-export OS_USER_DOMAIN_NAME=Default
-export OS_DOMAIN_ID=<domain_id_that_matches_name>
-export OS_PROJECT_NAME=<project_name>
-export OS_TENANT_NAME=<project_name>
-export OS_TENANT_ID=<project_id_that_matches_name>
-export OS_USERNAME=<username>
-export OS_PASSWORD=<password>
-export OS_AUTH_URL=http://<openstack_keystone_endpoint>/v3
-export OS_INTERFACE=public
-export OS_IDENTITY_API_VERSION=3
-export OS_REGION_NAME=<region_name>
-```
-
-The specific values you use will vary based on your particular environment. You may
-notice that several values are aliases of one another. This is in part because the
-values expected by the OpenStack client and
-[Gopher Cloud](https://github.com/gophercloud/gophercloud) are slightly different,
-especially with respect to the change from using `tenant` to `project`. One of our
-development goals is to make this setup easier and more consistent.
+openstack-cloud-controller-manager relies on [Octavia](https://docs.openstack.org/octavia/latest/) to create Kubernetes Service of type LoadBalancer, so make sure Octavia is available in the OpenStack cloud. For public cloud, you need to check the cloud documentation. For devstack, refer to Octavia deployment [quick start guide](https://docs.openstack.org/octavia/latest/contributor/guides/dev-quick-start.html).
```

Comment:
I knew Barbian is needed for TLS, do we want to mention it as optional?

Reply:
Good suggestion,